modules/core/api/tests/src/Kernel/FarmApiTest.php

```diff
@@ -78,26 +82,42 @@ public function setUp(): void {
     $this->setInstallProfile('farm');
 
     // Set the site name so that we can check for it in /api meta.farm info.
-    \Drupal::configFactory()->getEditable('system.site')->set('name', 'API Test')->save();
+    $this->config('system.site')->set('name', 'API Test')->save();
 
     // Allow JSON:API write operations and change the base path to /api.
     // These would normally be done by farm_api_install(), which does not run
     // in Kernel tests (it also does other things we don't need).
-    \Drupal::configFactory()->getEditable('jsonapi.settings')->set('read_only', FALSE)->save();
-    \Drupal::configFactory()->getEditable('jsonapi_extras.settings')->set('path_prefix', 'api')->save();
+    $this->config('jsonapi.settings')->set('read_only', FALSE)->save();
+    $this->config('jsonapi_extras.settings')->set('path_prefix', 'api')->save();
 
-    // Set up a user with the farm_manager role.
-    $user = $this->setUpCurrentUser([], [], FALSE);
-    $user->addRole('farm_manager');
+    // Configure static scopes.
+    $this->config('simple_oauth.settings')->set('scope_provider', 'static')->save();
+
+    // Configure client with password grant and no default scopes.
+    $this->client
+      ->set('scopes', [])
+      ->set('grant_types', ['password'])
+      ->save();
+
+    // Create a second user to ensure we are not using user id 1.
+    $this->user = $this->createUser();
+    $this->assertEquals(2, $this->user->id());
```

Comment:
This was needed because `simple_oauth` has special logic to grant all access for user number 1: https://git.drupalcode.org/project/simple_oauth/-/blob/2a553c1c1a6a4a269a056f493024034463331490/src/Authentication/TokenAuthUser.php#L90-93

I think simple_oauth will need to update this at some point. But for now this is an easy work around.